### DISCUSSION TITLE
In-app overview of findings per customer

### DISCUSSION THREAD

**Discussion by unixerius** [Syslifters/sysreptor#272] [General]
Tieing this suggestion in with the suggestion of having a Customer object -> https://github.com/Syslifters/sysreptor/discussions/36

It wouldn't just be great to group engagements/reports per customer. It would also be awesome if we can get an in-app overview of all pentest findings reported for this customer. That will make it easier to for example find everything we've ever reported, or to make a quick update to the right finding. 

Plus, management just really likes it when we can offer a most recent overview of which pentest findings we have and what their status is with the customer (if we know).

It also hinged on findings have a unique ID that auto-increments, which I believe was requested elsewhere.

In short a table per customer like:

|  Finding ID | Engagement / Project | Date reported | Type | Severity / CVSS | Summary description | Latest status |
----------------------------
|  MM-SS-2024-01 | Scary Shack | 01/01/2024 | Injection | 8.9 | Unauthenticated SQL injection on login page | Open |
|  MM-SS-2024-02 | Scary Shack | 01/01/2024 | Injection | 7.8 | Authenticated SQL injection on product search page | Open |

**Comment by aronmolnar** [comment]
Those are quite many smaller components or functionalities sticked together, and the problem is that all of them are quite specialized.

We will be working on a plugin system in H2 which might allow those features to still be implemented.

**Comment by unixerius** [comment]
That would be absolutely awesome. If you like I can split out this into multiple discussion items. Or would you like me to refine them in this thread into separate RFCs?

Unixerius B.V.

Bolderweg 2, kamer E.26, 1332 AT, Almere
KVK 63637456
________________________________

**Reply by aronmolnar** [reply]
No thats good as it is.  
We might split it in the future if parts are implemented.

**Comment by unixerius** [comment]
Awesome!

So far, your team has done nothing but impress my colleagues with your response times and effectiveness!

Unixerius B.V.

Bolderweg 2, kamer E.26, 1332 AT, Almere
KVK 63637456
________________________________

**Reply by aronmolnar** [reply]
Thank you! We're very happy to hear that.
